src/tests/nativeaot/SmokeTests/DotnetJs/Program.cs

```diff
@@ -44,5 +49,65 @@ internal static string Concat(string a, string b)
 
         [JSExport]
         internal static void Throw() => throw new Exception("This is a test exception");
+
+        [JSExport]
+        internal static async Task<int> Async(Task task, bool throwAfterAwait)
+        {
+            Console.WriteLine($"Async method started");
+            await task;
+            if (throwAfterAwait)
+                throw new Exception("Async throwAfterAwait");
+
+            Console.WriteLine("Async method completed");
+            return 87;
+        }
+
+        [JSExport]
+        internal static async Task<int> AsyncWithCancel()
+        {
+            Console.WriteLine("Fire HTTP");
+            using var cts = new CancellationTokenSource();
+            using var http = new HttpClient();
+            var responseTask = http.GetStringAsync("https://github.com", cts.Token);
```

Comment:
So even if the `Task.Delay` below is increased to something unreasonable like a minute, this is still expected to succeed?

Reply:
I'm fine removing the test. We don't support marshalling cancellation tokens and wiring them to `AbortController` ATM. So only HTTP and JS module import currently supports cancellation.

Reply:
Or we can dropt the delay.

Reply:
> So only HTTP and JS module import currently supports cancellation.

Ahh, so that's how things are.

My preference would be to find some HTTP entrypoint that never responds and make the test robust via that. Next option is keeping it as is, next is removing. Up to you.

Reply:
Found a super flow one